straxen/plugins/peaks/peak_se_score.py

```diff
@@ -0,0 +1,149 @@
+import numpy as np
+from numba import njit
+
+import strax
+import straxen
+
+
+class PeakSEScore(strax.OverlapWindowPlugin):
+    """This plugin is designed to calculate a score based on the position relations to single
+    electrons.
+
+    References:
+        * v0.2.0: xenon:xenonnt:analysis:hot_spot_cut_summary
+        * v0.4.0: xenon:xenonnt:ac:sr1:hotspot_veto_cut:wimp_roi
+        * position resolution: xenon:xenonnt:svetter:data_driven_resolution_sr1a0_vs_sr1a1
+
+    """
+
+    __version__ = "0.4.0"
+    depends_on = ("peak_basics", "peak_positions")
+    provides = "peak_se_score"
+    save_when = strax.SaveWhen.EXPLICIT
+
+    dtype = strax.time_fields + [
+        (
+            ("Sum of the PDF of the SE position nearby probability.", "se_score"),
+            np.float32,
+        ),
+    ]
+
+    se_time_search_window_left = straxen.URLConfig(
+        default=5e9, type=int, help="SEs time window searching window, left side [ns]"
+    )
+
+    se_time_search_window_right = straxen.URLConfig(
+        default=0,
+        type=int,
+        help="Time searching window right side extension, one max drift time [ns]",
+    )
+
+    se_selection_area_roi = straxen.URLConfig(
+        default=[10, 80],
+        type=list,
+        help="Area range for single electron selection.[PE]",
+    )
+
+    se_selection_width_roi = straxen.URLConfig(
+        default=[80, 700],
+        type=list,
+        help="Area range for single electron selection.[PE]",
+    )
+
+    para_a_naive = straxen.URLConfig(
+        default=1,
+        help="Place holder for Parameter A in the position resolution function.",
+    )
+
+    para_b_naive = straxen.URLConfig(
+        default=1,
+        help="Place holder for Parameter B in the position resolution function.",
+    )
+
+    def get_window_size(self):
+        # This method is required by the OverlapWindowPlugin class
+        return 2 * (self.se_time_search_window_left + self.se_time_search_window_right)
+
+    def setup(self):
+        self._para_a = self.para_a_naive
+        self._para_b = self.para_b_naive
+
```

Comment:
Hey @chnlkx, sorry for the dummy question, but is this necessary?

Reply:
Hi @GiovanniVolta, the parameters a and b change between phase_0 and phase_1, but this information could only be implemented in cutax. So we put the phase-dependent parameters in https://github.com/XENONnT/cutax/pull/479, which will override the para_a and para_b here. So here they are more like a naive placeholder.